fontforge/python.c

```diff
@@ -17937,22 +17937,43 @@ return (NULL);
 Py_RETURN( self );
 }
 
+/* Transformation flags: see 'enum fvtrans_flags' in baseviews.h */
+static struct flaglist f_trans_flags[] = {
+    { "activeLayer", fvt_alllayers },
+    { "guide", fvt_dogrid },
+    { "noWidth", fvt_dontmovewidth },
+    { "round", fvt_round_to_int },
+    { "simplePos", fvt_scalepstpos },
+    { "kernClasses", fvt_scalekernclasses },
+    FLAGLIST_EMPTY /* Sentinel */
+};
+
 static PyObject *PyFFFont_Transform(PyFF_Font *self, PyObject *args) {
     FontViewBase *fv;
     int i;
     double m[6];
     real t[6];
     BVTFunc bvts[1];
+    int flags;
+    PyObject *flagO=NULL;
 
     if ( CheckIfFontClosed(self) )
 return (NULL);
     fv = self->fv;
-    if ( !PyArg_ParseTuple(args,"(dddddd)",&m[0], &m[1], &m[2], &m[3], &m[4], &m[5] ) )
+    if ( !PyArg_ParseTuple(args,"(dddddd)|O",&m[0], &m[1], &m[2], &m[3], &m[4], &m[5], &flagO ) )
 return( NULL );
+    flags = FlagsFromTuple(flagO,f_trans_flags,"transformation flag");
+    if ( flags==FLAG_UNKNOWN )
+return( NULL );
+    // Flip fvt_alllayers back
+    if ( flags&fvt_alllayers )
+	flags |= fvt_alllayers;
```

Comment:
Why is this needed? Isn't this a no-op?

Reply:
The intention was to have fvt_alllayers in the python flags be opposite to that in the flags passed down (as its supposed to be on by default). I'll take a look later.

Reply:
It's going to be Wed evening at the earliest before I could do something about this properly, but I believe the line should read `flags &= ~fvt_alllayers;` . I just goofed it, apparently. If that makes sense to you and you want to integrate early, go ahead.

Reply:
OK, had time to actually look at and revise this. I just had the two bit-flipping lines in the wrong positions.